conn.go

```diff
@@ -1662,16 +1662,26 @@ func (c *Conn) query(ctx context.Context, statement string, values ...interface{
 
 func (c *Conn) awaitSchemaAgreement(ctx context.Context) (err error) {
 	const (
-		peerSchemas  = "SELECT * FROM system.peers"
-		localSchemas = "SELECT schema_version FROM system.local WHERE key='local'"
+		peerSchemas   = "SELECT * FROM system.peers"
+		peerSchemasV2 = "SELECT * FROM system.peers_v2"
+		localSchemas  = "SELECT schema_version FROM system.local WHERE key='local'"
 	)
 
 	var versions map[string]struct{}
 	var schemaVersion string
 
 	endDeadline := time.Now().Add(c.session.cfg.MaxWaitSchemaAgreement)
+
+	var queryString string
+
+	if c.host.version.AtLeast(4, 0, 0) {
+		queryString = peerSchemasV2
+	} else {
+		queryString = peerSchemas
+	}
```

Comment:
This version check is in three places. Perhaps we should have a function that returns peers table name based on the version?

```go
func peersTableName(version cassVersion) string {
	if version.AtLeast(4, 0, 0) {
		return "system.peers_v2"
	}
	return "system.peers"
}
```

Reply:
added.